include/pthread.h

```diff
@@ -195,11 +195,8 @@
  * error number in errno.  This is an inconsistency with the pthread interfaces.
  */
 
-#define pthread_setname_np(thread, name) \
-  prctl((int)PR_SET_NAME_EXT, (char*)name, (int)thread)
-
-#define pthread_getname_np(thread, name) \
-  prctl((int)PR_GET_NAME_EXT, (char*)name, (int)thread)
+#define pthread_setname_np(thread, name) prctl(PR_SET_NAME_EXT, name, thread)
+#define pthread_getname_np(thread, name) prctl(PR_GET_NAME_EXT, name, thread)
```

Comment:
is this change necessary? i feel it's safer to be explicit about types for vararg functions.

Reply:
Ok, I will drop the last commit from PR